newapi/ooniapi/measurements.py

```diff
@@ -1056,6 +1049,37 @@ def param_url(name):
     return p
 
 
+def param_report_id_or_none() -> Optional[str]:
+    p = request.args.get("report_id")
+    if not p:
+        return None
+    if len(p) < 15 or len(p) > 100:
+        raise BadRequest(f"Invalid report_id field")
+    accepted = string.ascii_letters + string.digits + "_"
+    validate(p, accepted)
+    return p
+
+
+def param_report_id() -> str:
+    rid = param_report_id_or_none()
+    if rid is None:
+        raise BadRequest(f"Invalid report_id")
```

Comment:
I would consider returning a less informative err message (maybe an error code?).

Reply:
It is bubbling up by design to be able to show what field is incorrect. I don't think adding opacity here would be useful because the source code is public.